packages/NodeTypeResolver/TypeComparator/TypeComparator.php

```diff
@@ -247,4 +255,11 @@ private function normalizeConstantBooleanType(Type $type): Type
             return $callable($type);
         });
     }
+
+    private function isTypeSelfAndDocParamTypeStatic(Type $phpStanDocType, Type $phpParserNodeType, TypeNode $phpStanDocTypeNode) : bool
+    {
+        return $phpStanDocType instanceof StaticType
+            && $phpParserNodeType instanceof ThisType
```

Comment:
I think `in_array()` can be used so we have specific types to check:

```suggestion
        return in_array($phpStanDocType::class, [StaticType::class, ThisType::class], true)
```

Reply:
Can you specify what you mean? I'm not sure what part of condition do you want to replace by it, I use `$phpStanDocType ~ StaticType` and `$phpParserNodeType ~ ThisType`

Reply:
Oops, you're correct, it is ok then 👍